lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -392,17 +411,40 @@ def dismiss_welcome_message(request):
 def save_course_goal(request):
     course_id = request.data.get('course_id', None)
     goal_key = request.data.get('goal_key', None)
+    number_of_days_with_visits_per_week_goal = request.data.get('number_of_days_with_visits_per_week_goal', None)
+    subscribed_to_goal_reminders = request.data.get('subscribed_to_goal_reminders', None)
 
     # If body doesn't contain 'course_id', return 400 to client.
     if not course_id:
         raise ParseError(_("'course_id' is required."))
 
-    # If body doesn't contain 'goal', return 400 to client.
-    if not goal_key:
-        raise ParseError(_("'goal_key' is required."))
+    if COURSE_GOALS_NUMBER_OF_DAYS_GOALS.is_enabled():
+        # If body doesn't contain the required goals fields, return 400 to client.
+        if number_of_days_with_visits_per_week_goal is None or subscribed_to_goal_reminders is None:
+            raise ParseError(_("'number_of_days_with_visits_per_week_goal' and 'subscribed_to_goal_reminders' are required."))
+
+        try:
+            number_of_days_with_visits_per_week_goal = int(number_of_days_with_visits_per_week_goal)
+        except ValueError:
+            raise ParseError(_("'number_of_days_with_visits_per_week_goal' needs to be an integer"))
+        subscribed_to_goal_reminders = subscribed_to_goal_reminders == 'True'
```

Comment:
it is expecting the frontend to send subscribed_to_goal_reminders as the string True or False

Reply:
These inputs don't get just get parsed as json? Might be easier for you and more readable that way. Should be able to find examples of other endpoints doing that, I thought that's how most of em worked?

Reply:
i tried using request.json and the expect_json decorator locally like I saw in some other places in platform, but i think there might be some headers or something missing from the existing frontend for that to work, so it might make sense to wait for the clean up ticket to try to switch to that

Reply:
Ah - the thing I think you are missing is using `content_type='application/json'` in your test `post()` call. Django is ancient and still defaults to using `multipart/form-data` by default. I believe our MFE ajax calls use json by default, so best to match that anyway.

Once you set that content_type, it looks like you probably don't need `expect_json` decorator or `request.json` -- `request.data` looks like it will be parsed correctly for you

Reply:
hm, using ensure_json and request.json wasn't working for me locally for some reason but like you pointed out with the right headers request.data already works correctly, so updated the code to reflect that